src/socket.cpp

```diff
@@ -33,35 +52,54 @@ static inline void setNoDelay(int socket) {
 }
 
 static inline void setQuickAck(int socket) {
+#ifndef _WIN32
 #ifdef TCP_QUICKACK
     int value = 1;
     if (setsockopt(socket, IPPROTO_TCP, TCP_QUICKACK, (char*)&value, sizeof(int)) < 0)
         throw std::runtime_error("Error setting quick ack");
 #endif
+#endif
 }
 
 static inline void writeSocket(int socket, const void* data, size_t size) {
     while (size > 0) {
-        int s = send(socket, (char*)data, size, 0);
+        int s = send(socket, (const char*)data, size, 0);
         if (s < 0) {
+#ifdef _WIN32
+            if (WSAGetLastError() == WSAEWOULDBLOCK) {
+                continue;
+            }
+#else
             if (SOCKET_LAST_ERRCODE == EAGAIN) {
                 continue;
             }
-            throw WriteSocketException(SOCKET_LAST_ERRCODE, SOCKET_LAST_ERROR);
+#endif
+            throw std::runtime_error("Error writing to socket");
         } else if (s == 0) {
-            throw WriteSocketException(0, "Socket closed");
+            throw std::runtime_error("Socket closed");
         }
         size -= s;
-        data = (char*)data + s;
+        data = (const char*)data + s;
     }
 }
 
 static inline bool tryReadSocket(int socket, void* data, size_t size, unsigned long maxAttempts) {
     // maxAttempts = 0 means infinite attempts
     size_t s = size;
     while (s > 0) {
-        int r = recv(socket, data, s, 0);
+        int r = recv(socket, (char*)data, s, 0);
         if (r < 0) {
+#ifdef _WIN32
+            if (WSAGetLastError() == WSAEWOULDBLOCK) {
```

Comment:
This could be an inline function:

```
if (WSAGetLastError() == WSAEWOULDBLOCK)
```

```
if (SOCKET_LAST_ERRCODE == EAGAIN)
```

like `isEgainError()`.

Reply:
Done :)